### PR TITLE
Feature/19 integration tests

### PR DESCRIPTION
**Pull request by pj-spoelders**
Adds:
#19 - integration tests
#18 - OIDC integration
#17 - custom query

Also cleaned up the pq file significantly, added some more comments, simplified logic

### CODE REVIEW

File: UIAutomationTests/TestsUsernamePassword.cs [hunk 1]
```diff
@@ -0,0 +1,331 @@
+﻿using System;
+using System.IO;
+using Xunit;
+
+namespace UIAutomationTests
+{
+    //https://xunit.net/docs/running-tests-in-parallel
+    //We don't want to run them in parallel (since we're doing automation UI testing) so we add the tests to the same test collection
+    [Collection("NotParallel")]
+    public class TestsUsernamePassword : IClassFixture<TestFixture>
```
Comment (chiaradiamarcelo) [inline UIAutomationTests/TestsUsernamePassword.cs:10]: would be nice to separate the tests case that are not authentication-specific to another class

File: UIAutomationTests/TestsKeyOIDC.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            Assert.True(String.IsNullOrWhiteSpace(error), $@"Errormessage: {error}");
+            Assert.True(grid.RowCount == 5 + 1, $@"actual rowCount is {grid.RowCount}");
+        }
+        [Fact]
+        public void OdbcDatasourceAW()
+        {
+            string MQueryExpression = File.ReadAllText("QueryPqFiles/ExasolAW.query.pq");
+
+            var (error, grid) = testFixture.Test(MQueryExpression);
+
```
Comment (chiaradiamarcelo) [inline UIAutomationTests/TestsKeyOIDC.cs:35]:
```suggestion
```

File: UIAutomationTests/TestsKeyOIDC.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+namespace UIAutomationTests
+{
+    [Collection("NotParallel")]
+    public class TestsKeyOIDC : IClassFixture<TestFixture>
+    {
+        TestFixture testFixture;
+        public TestsKeyOIDC(TestFixture tf)
+        {
+            testFixture = tf;
+            tf.Authenticate(TestFixture.AuthenticationMethod.KeyOIDCToken);
```
Comment (chiaradiamarcelo) [inline UIAutomationTests/TestsKeyOIDC.cs:16]: would be nice to un-authenticate when all tests in the class are done.

Ideally would be nice to have the authentication/un-authentication in every test, so every test starts with a clean state and leaves everything the way it was after its execution.

File: UIAutomationTests/readme.md [hunk 4]
```diff
@@ -12,5 +12,11 @@ Provides a way of automatically testing the Exasol PowerBI Connector
...
-- provide configuration in a config file eventually
\ No newline at end of file
+- Elaborate tests (highest prio) (WIP)
+- provide configuration in a config file eventually (done)
+
+## Data used for the integration tests
+- adventureworks conversion by T.B.
+- "TEST": an empty schema
+- the originally named schema 'MORETESTS' described in MORETESTS.SQL
+
```
Comment (chiaradiamarcelo) [inline UIAutomationTests/readme.md:22]: looks like there's an `documentation/New Text Document.txt ` empty file

**Review by chiaradiamarcelo — APPROVED** [review]

File: UIAutomationTests/TestsKeyOIDC.cs (hunk 3 above)
Comment (pj-spoelders) [reply UIAutomationTests/TestsKeyOIDC.cs:16]: I fully agree. I'll alter it. It was a shortcut back then to get things going.

File: UIAutomationTests/TestsUsernamePassword.cs (hunk 1 above)
Comment (pj-spoelders) [reply UIAutomationTests/TestsUsernamePassword.cs:10]: I agree. I'll split them up.

File: UIAutomationTests/moretests.sql [hunk 5]
```diff
@@ -0,0 +1,42 @@
+﻿CREATE SCHEMA TEST;
+CREATE SCHEMA MORETESTS;
+CREATE TABLE MORETESTS.MORETYPES (
```
[the remark this answers is not in the document]
Comment (pj-spoelders) [reply UIAutomationTests/moretests.sql]: Done

**Pull request merged**